roborock/web_api.py

```diff
@@ -332,6 +332,29 @@ async def get_home_data_v2(self, user_data: UserData) -> HomeData:
         else:
             raise RoborockException("home_response result was an unexpected type")
 
+    async def get_home_data_v3(self, user_data: UserData) -> HomeData:
+        """This is the same as get_home_data, but uses a different endpoint and includes non-robotic vacuums."""
+        rriot = user_data.rriot
+        if rriot is None:
+            raise RoborockException("rriot is none")
+        home_id = await self._get_home_id(user_data)
+        if rriot.r.a is None:
+            raise RoborockException("Missing field 'a' in rriot reference")
```

Comment:
Swap order so the two rriot checks are nearby?

```suggestion
        if rriot.r.a is None:
            raise RoborockException("Missing field 'a' in rriot reference")
        home_id = await self._get_home_id(user_data)
```

Reply:
I removed one rriot check as I am instead setting the value to not optional.